pkg/cmd/codespace/common.go

```diff
@@ -59,6 +61,37 @@ func (a *App) StopProgressIndicator() {
 	a.io.StopProgressIndicator()
 }
 
+// Connects to a codespace using Live Share and returns that session
+func startLiveShareSession(ctx context.Context, codespace *api.Codespace, a *App, debug bool, debugFile string) (session *liveshare.Session, err error) {
+	// While connecting, ensure in the background that the user has keys installed.
+	// That lets us report a more useful error message if they don't.
+	authkeys := make(chan error, 1)
+	go func() {
+		authkeys <- checkAuthorizedKeys(ctx, a.apiClient)
+	}()
+
+	liveshareLogger := noopLogger()
+	if debug {
+		debugLogger, err := newFileLogger(debugFile)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't create file logger: %w", err)
+		}
+		defer safeClose(debugLogger, &err)
+
+		liveshareLogger = debugLogger.Logger
+		a.errLogger.Printf("Debug file located at: %s", debugLogger.Name())
+	}
+
+	session, err = codespaces.ConnectToLiveshare(ctx, a, liveshareLogger, a.apiClient, codespace)
+	if err != nil {
+		if authErr := <-authkeys; authErr != nil {
+			return nil, fmt.Errorf("failed to fetch authorization keys: %w", authErr)
+		}
+		return nil, fmt.Errorf("failed to connect to Live Share: %w", err)
+	}
+	return
```

Comment:
I was out half of last week so didn't get a chance to review this again before it merged. My comment about adding the return named value to safeClose still stands but one of the unfortunate warts of named return values is that you can do empty returns like this, which is frowned upon.

I'd recommend changing this to: `return session, nil`

Reply:
https://github.com/cli/cli/pull/5568